tests/ArmTemplates.Tests/Extractor/Scenarios/NamedValuesExtractorTests.cs

```diff
@@ -98,5 +98,56 @@ public async Task GenerateNamedValuesTemplates_ProperlyLaysTheInformation()
             namedValuesTemplate.TypedResources.NamedValues.First().Name.Should().Contain(MockNamedValuesClient.NamedValueName);
             namedValuesTemplate.TypedResources.NamedValues.First().Properties.DisplayName.Should().Contain(MockNamedValuesClient.NamedValueDisplayName);
         }
+
+        [Fact]
+        public async Task GenerateNamedValuesTemplates_ProperlyLaysTheInformation_GivenExtractSecretSetToTrue()
+        {
+            // arrange
+            var currentTestDirectory = Path.Combine(this.OutputDirectory, nameof(GenerateNamedValuesTemplates_ProperlyLaysTheInformation_GivenExtractSecretSetToTrue));
+
+            var extractorConfig = this.GetDefaultExtractorConsoleAppConfiguration(extractSecrets: "true");
+            var extractorParameters = new ExtractorParameters(extractorConfig);
+
+            var fileLocationNamedValues = Path.Combine(MockClientUtils.ApiClientJsonResponsesPath, "ApiManagementListNamedValues_success_response.json");
```

Comment:
nit: maybe in future PRs we can extract names of json files into separate class so that we can track their usage somehow. Dont mind changing it in current PR

Reply:
Agree will be useful, thank you! Will update in upcoming PR with tests